File: Loenn/entities/disableUpTransitionController.lua
-- Copyright (c) Shane Woolcock. Licensed under the MIT Licence.
-- See the LICENCE file in the repository root for full licence text.

local consts = require("mods").requireFromPlugin("consts")
local helpers = require("mods").requireFromPlugin("helpers")

local placementData = helpers.createPlacementData('1', {
})

local disableUpTransitionController = {
    name = "GravityHelper/DisableUpTransitionController",
    depth = -8500,
    texture = "objects/GravityHelper/gravityController/noup",
    ignoredFields = consts.ignoredFields,
    placements = {
        {
            name = "normal",
            data = helpers.union(placementData),
        },
    },
}

return disableUpTransitionController

File: Source/Entities/Controllers/DisableUpTransitionController.cs
// Copyright (c) Shane Woolcock. Licensed under the MIT Licence.
// See the LICENCE file in the repository root for full licence text.

using Celeste.Mod.Entities;
using Microsoft.Xna.Framework;
using Monocle;

namespace Celeste.Mod.GravityHelper.Entities.Controllers;

[Tracked]
[CustomEntity("GravityHelper/DisableUpTransitionController")]
public class DisableUpTransitionController : Entity
{
    public DisableUpTransitionController(EntityData data, Vector2 offset) : base(data.Position + offset)
    {
    }
}
